### PR TITLE
Setup classes

### PR DESCRIPTION
**Pull request by AmanWorku**
### Hello

# Activities done:
- [x] Configure linters.
- [x] Created `Person.rb` with respective instances. 
- [x] Created `Student.rb` with respective instances and added a method. 
- [x] Created `Teacher.rb` with respective instances and overridden a method inherited from the Person class.

### CODE REVIEW

**Comment by omarsalem7** [comment]
Hi @AmanWorku 😊, 

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request 🛑 💀 

You have submitted a project, that doesn't have a descriptive readme, kindly update it and make sure to include all sections in the readme

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by AmanWorku** [comment]
@omarsalem7 My bad sir. I will add it now. Thank you.

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    of_age? || @parent_permission
+  end
+end
+
+private :is_of_age
```
Comment (Wes-Isaac) [inline person.rb:23]: - [ ] Great job so far, when I ran this code it gave a NoMethodError the method is_of_age is not defined, let's fix the typo and make sure we make the method private.

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+class Person
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  attr_accessor :age, :name
+  attr_reader :id
```
Comment (Wes-Isaac) [inline person.rb:12]: - [ ] [OPTIONAL] It would be better if we define the accessor and reader right after the class.

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @AmanWorku, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Professionally documented your work 😮.
- Great communication with the reviewer 👍.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by AmanWorku** [comment]
@Wes-Isaac Thank you brother. Amesegenalew.

**Review by prolajumokeoni — APPROVED** [review]
Hi @AmanWorku 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- All changes requested have been implemented 👏🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AmanWorku** [comment]
@prolajumokeoni Thank you so much.

**Pull request merged**